### PR TITLE
HierarchyId support #339

### PR DESCRIPTION
**Pull request by rbeurskens**
Migrated @zgabi 's [fork on codeplex](https://entityframework.codeplex.com/SourceControl/network/forks/zgabi/efhierarchyid) (many thanks) which (I think) is the source of the NuGet Package [EntityFrameworkWithHierarchyId](https://www.nuget.org/packages/EntityFrameworkWithHierarchyId/) to GitHub and reverted changes related to changing the public assembly key and NuGet package Id. 

As stated in feature request #339, community contributions for features that the EF team is not planning to implement themselves can still be considered as most of it would be already done.

( I am not the original author, I am only trying to add this contribution to the EF code base so it has the possibility to support the data type without any hacks/workarounds. It would be a shame if it is lost when CodePlex goes down. )

### CODE REVIEW

File: samples/Provider/SampleEntityFrameworkProvider/ProviderManifest.cs [hunk 1]
```diff
@@ -0,0 +1,617 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.
```
Comment (zgabi) [inline samples/Provider/SampleEntityFrameworkProvider/ProviderManifest.cs]: This seems to be a merge mistake. This is not my file. Ok, it was modified by me, but it seems that later it was removed. So you should remove this file from the commit (and the following 2 files, too)

**Comment by zgabi** [comment]
@rbeurskens I've updated my fork. Now it is on github, too. Could you please recreate from this fork?
https://github.com/zgabi/EntityFramework6/commits/master

Not only merged the latest upstream code. HierarchyID changes are based on spatial (geography, geomerty) codes, which was already supproted by EF. It was changed in the last years, and now I updated the same changes in the HierarchyID classes. For example "annotations" parameter was added to the ColumnBuilder, and some code comment style was also changed, so I make it similar:)

I separated the changes to 2 commits, you need to make a PR from this state:
https://github.com/zgabi/EntityFramework6/tree/f86943e1e3b2cfc0b3e7fb8ba4334bff95f0567b

The next commit is the "Package rename and assembly signing changes", it it not needed.

**Comment by rbeurskens** [comment]
@zgabi I pulled your fork and rebased this PR branch on it ( replacing the original commits with the one you linked )

**Comment by rbeurskens** [comment]
What else needs to happen before this can be merged? I still need to do anything?

**Comment by ajcvickers** [comment]
@rbeurskens Sorry this has taken a long time to get to. The team is currently focused on the EF Core 2.1 release. As that work wraps up we have some EF Designer work to do, and as soon as that is complete we will start working on the next EF6 release, including reviewing community pull requests.

**Comment by rbeurskens** [comment]
Thanks for your response. I understand. I Just wanted to make sure I didn't forget something.

**Comment by millsy013** [comment]
Good day, how goes the development of this feature? Is there anything I can do to help speed along the process?

**Comment by rbeurskens** [comment]
@millsy013 --Development-- has been completed by @zgabi .
It is now a matter of review and approval by the EF team so it can be included in the next EF6.x release, however they have other priorities right now. (as explained by @ajcvickers) I don't think we as community can do anything in this process as it can be seen as a final quality check by Microsoft.
I don't have a clue for an ETA for the next (official) EF6.x version on NuGet, but the good news is: there will be one and the project is still maintained. All we can do for now is wait, I guess.

**Comment by rbeurskens** [comment]
@ajcvickers I see there is a lot of activity on issues for the next release going on. Could you please re-open the issue #339 related to this PR so we can get in indication if/when this is considered to be reviewed/included in a release? (currently it is just closed without a milestone or any status flag)
Thanks.

**Comment by ajcvickers** [comment]
This has now been merged. Thank you for your patience!

**Comment by rbeurskens** [comment]
Thank you. Apologies for pushing it a bit.

**Comment by zgabi** [comment]
Thank you.

**Pull request closed without merging**